app/api/orders.py

```diff
@@ -385,6 +389,22 @@ def before_update_object(self, order, data, view_kwargs):
             raise UnprocessableEntity({'pointer': '/data/attributes/payment-mode'},
                                       "insufficient data to verify paypal payment")
 
+        status = data.get('status', None)
+        if status and status not in ['cancelled', 'completed', 'expired', 'initializing', 'pending', 'placed']:
+            raise ConflictException({'pointer': '/data/attributes/status'},
+                                    "status {} is invalid".format(status))
+        elif status and status != 'cancelled':
+            if (order.status in ['completed', 'expired', 'cancelled'] and order.status != status) or \
+                    (order.status == 'placed' and status not in ['completed', 'placed']) or \
+                    (order.status == 'pending' and status not in ['completed', 'expired', 'pending']) or \
+                    (order.status == 'initializing' and status not in ['completed', 'expired', 'pending', 'placed']):
+                raise ConflictException({'pointer': '/data/attributes/status'},
```

Comment:
Regarding 3rd point, we cannot update the status of expired order to any other status. Neither cancelled nor any other @iamareebjamal @kushthedude 
However keeping second point in my mind, I'll add checks to avoid status transition for deleted orders

Reply:
@kushthedude were you able to change status of expired orders too because I think there is a check to avoid that